packages/flutter/lib/src/cupertino/button.dart

```diff
@@ -9,21 +9,85 @@ import 'package:flutter/foundation.dart';
-// outline color.
-const double _kCupertinoFocusColorOpacity = 0.80;
-const double _kCupertinoFocusColorBrightness = 0.69;
-const double _kCupertinoFocusColorSaturation = 0.835;
+// Measured against iOS 17 guidelines (https://developer.apple.com/design/human-interface-guidelines/buttons#iOS-iPadOS)
+
+/// The size of a [CupertinoButton].
+/// Based on the iOS (17) Human Interface Guidelines (https://developer.apple.com/design/human-interface-guidelines/buttons#iOS-iPadOS)
+enum CupertinoButtonSize {
+  /// Displays a smaller button with round sides and smaller text (uses [CupertinoTextThemeData.actionSmallTextStyle]).
+  small,
+  /// Displays a medium sized button with round sides and regular-sized text
+  medium,
+  /// Displays a (classic) large button with rounded edges and regular-sized text
+  large,
+}
+
+/// The padding values for the different [CupertinoButtonSize]s.
+/// Based on the iOS (17) Human Interface Guidelines (https://developer.apple.com/design/human-interface-guidelines/buttons#iOS-iPadOS)
+const Map<CupertinoButtonSize, EdgeInsetsGeometry> kCupertinoButtonPadding = <CupertinoButtonSize, EdgeInsetsGeometry>{
+  CupertinoButtonSize.small: EdgeInsets.symmetric(
+    vertical: 6,
+    horizontal: 12,
+  ),
+  CupertinoButtonSize.medium: EdgeInsets.symmetric(
+    vertical: 10,
+    horizontal: 15,
+  ),
+  CupertinoButtonSize.large: EdgeInsets.symmetric(
+    vertical: 16,
+    horizontal: 20,
+  ),
+};
+
+/// The border radius values for the different [CupertinoButtonSize]s.
+/// Based on the iOS (17) Human Interface Guidelines (https://developer.apple.com/design/human-interface-guidelines/buttons#iOS-iPadOS)
+final Map<CupertinoButtonSize, BorderRadius> kCupertinoButtonSizeBorderRadius = <CupertinoButtonSize, BorderRadius>{
+  CupertinoButtonSize.small: BorderRadius.circular(40),
+  CupertinoButtonSize.medium: BorderRadius.circular(40),
+  CupertinoButtonSize.large: BorderRadius.circular(12),
+};
+
+/// The minimum size of a [CupertinoButton] based on the [CupertinoButtonSize].
+/// Based on the iOS (17) Human Interface Guidelines (https://developer.apple.com/design/human-interface-guidelines/buttons#iOS-iPadOS)
+const Map<CupertinoButtonSize, double> kCupertinoButtonMinSize = <CupertinoButtonSize, double>{
+  CupertinoButtonSize.small: 28,
+  CupertinoButtonSize.medium: 32,
+  CupertinoButtonSize.large: 44,
+};
+
+
+
+/// The style of a [CupertinoButton] that changes the style of the button's background.
+/// Based on the iOS Human Interface Guidelines (https://developer.apple.com/design/human-interface-guidelines/buttons#iOS-iPadOS)
+enum _CupertinoButtonStyle {
+  /// No background or border, primary foreground color
+  plain,
+  /// Translucent background, primary foreground color
+  tinted,
+  /// Solid background, contrasting foreground color
+  filled,
+}
+
+
+
+/// The relative values needed to transform a color to it's equivalent focus outline color.
+const double kCupertinoFocusColorOpacity = 0.80,
```

Comment:
These constants are used everywhere for focus in cupertino: switch, checkbox, radio, button, etc. If we are making them public, they should probably be defined in one place (possibly `constants.dart`) and made public there instead.

Reply:
I'm wondering in general where these public constants should live. Maybe the ones directly relevant to CupertinoButton should live here, and everything else move to cupertino/constants.dart?

Reply:
Opened https://github.com/flutter/flutter/pull/153115 to kickstart this move.